lib/cucumber/glue/proto_world.rb

```diff
@@ -92,7 +92,7 @@ def attach(file, media_type = nil)
         return super unless File.file?(file)
 
         content = File.read(file, mode: 'rb')
-        media_type = MIME::Types.type_for(file).first if media_type.nil?
+        media_type = MiniMime.lookup_by_filename(file)&.content_type if media_type.nil?
```

Comment:
If you look above on line 82 you'll see a very common media type that we log is a custom one we've invented. I'd be surprised if this was one of the cached ones in mini-mime (The docs there state that this new gem is a lot slower for uncached ones).

Reply:
FYI: the `MiniMime.lookup_by_filename` call is only made if a `media_type` value _isn't_ provided to the `attach` method. It's there to fill in the gap if the type isn't known at ahead of time.

The example on line 82 will not result in a call to `MiniMime.lookup_by_filename`.

Reply:
I believe this is the use case:

https://github.com/cucumber/cucumber-ruby/blob/cd5f138416bd3307c406c0720ba04ba75cd6c123/features/docs/writing_support_code/attachments.feature#L33-L35